eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/DefaultEventStoreTransaction.java

```diff
@@ -60,11 +60,15 @@ public class DefaultEventStoreTransaction implements EventStoreTransaction {
      *                           {@link #appendEvent(EventMessage) append events} with.
      * @param processingContext  The {@link ProcessingContext} from which to
      *                           {@link #appendEvent(EventMessage) append events} and attach resources to.
+     * @param tagResolver        The {@link TagResolver} used to resolve tags while
+     *                           {@link #appendEvent(EventMessage) appending events}.
      */
     public DefaultEventStoreTransaction(@Nonnull AsyncEventStorageEngine eventStorageEngine,
-                                        @Nonnull ProcessingContext processingContext) {
+                                        @Nonnull ProcessingContext processingContext,
+                                        @Nonnull TagResolver tagResolver) {
```

Comment:
I'm wondering if we should put all this logic in the Transaction class. We should consider adding 3 functions here: `onPrepareCommit`, `onCommit`, `onRollback`, where the EventStore can register the logic to be executed in the various stages of the transaction.

Reply:
Created issue for that:

https://github.com/AxonFramework/AxonFramework/issues/3253